tests/basic/storage_pool_test.go

```diff
@@ -10692,6 +10830,10 @@ var _ = Describe("{HAIncreasePoolresizeAndAdddisk}", func() {
 
 	itLog := "HAIncreasePoolresizeAndAdddisk"
 	It(itLog, func() {
+		isPoolAddDiskSupported, disk_err := IsPoolAddDiskSupported()
+		if !isPoolAddDiskSupported {
+			log.FailOnError(disk_err, "Add disk operation is not supported")
+		}
```

Comment:
How about?

```go
isPoolAddDiskSupported, err := IsPoolAddDiskSupported()
log.FailOnError(err, "failed to determine if pool add disk is supported")
if !isPoolAddDiskSupported {
	disk_err := fmt.Errorf("drive add to existing pool not supported for px-storev2 or px-cache pools")
	log.FailOnError(disk_err, "Add disk operation is not supported")
}
```

Reply:
This is not required krishna , because our function will return same log message you have put for fmt.Error if the condition doesnot satisfy

Reply:
If I plan to use IsPoolAddDiskSupported for another purpose, I would expect it to return either true or false. If it returns an error, I would assume it failed to determine if add disk is supported. I feel, returning an error along with false reduces its usability.